src/celeritas/phys/PhysicsStepUtils.hh

```diff
@@ -194,9 +195,22 @@ calc_mean_energy_loss(ParticleTrackView const& particle,
     {
         auto grid_id = physics.value_grid(VGT::energy_loss, ppid);
         CELER_ASSERT(grid_id);
-        auto calc_eloss_rate
-            = physics.make_calculator<EnergyLossCalculator>(grid_id);
-        eloss = Energy{step * calc_eloss_rate(pre_step_energy)};
+
+        size_type order = physics.scalars().spline_eloss_order;
+        CELER_ASSERT(order > 0);
+
+        if (order == 1)
+        {
+            auto calc_eloss_rate
+                = physics.make_calculator<XsCalculator>(grid_id);
+            eloss = Energy{step * calc_eloss_rate(pre_step_energy)};
+        }
+        else if (order > 1)
+        {
```

Comment:
This should probably be:
```suggestion
        else
        {
```
unless you want to add an `else { CELER_ASSERT_UNREACHABLE(); }` after the `else`.